### DISCUSSION TITLE
Adding pydantic as a requirement

### DISCUSSION THREAD

**Discussion by afrankenthal** [cms-nanoAOD/correctionlib#32] [Q&A] [answered]
Hi Nick,

I have adapted our Muon POG framework to produce the V1 schema jsons (you can see it here: https://gitlab.cern.ch/cms-muonPOG/spark_tnp/-/blob/master/prepare.py#L567-613). This works and I have added correctionlib as a submodule so I can use correctionlib.correctionlib.schemav1 directly, but I forgot that this also requires pydantic, which I had manually installed. Do you have suggestions on how to include pydantic as well? Or is this something you could include as a dependency in correctionlib, in addition to the other 3 repos? I tried adding pydantic as a submodule too, but the relative paths between schemav1 and pydantic are not correct. I'm trying to get to a point where people can just clone the POG repository and all needed submodules, and start running, if possible.

Thanks for any tips.
Cheers,
Andre

**Comment by nsmith-** [comment, marked as answer]
As of v1.1 (just yesterday) I restructured the repository to be a proper python package with pydantic as a dependency. You should now be able to `pip3 install correctionlib==1.1` instead of cloning. Let me know how it goes.

**Reply by afrankenthal** [reply]
Thanks for that info, I was able to install via pip without issues, and updated my repo to remove correctionlib as a submodule and instruct users to pip install it instead. Pydantic was successfully installed as a dependency too. I also changed the code to check if correctionlib is installed before trying to produce schema jsons (for now I'm making it only recommended, not required): https://gitlab.cern.ch/cms-muonPOG/spark_tnp/-/blob/master/prepare.py#L571-581

Thanks for your help,
Andre
